src/pages/login/login.ts

```diff
@@ -49,22 +55,33 @@ export class LoginPage extends BasePageComponent {
 
   login = (): Promise<any> =>
     this.platform.ready()
-      .then(() =>
-        this.authenticationProvider
-          .login()
-          .then(() => this.appConfigProvider.loadRemoteConfig())
-          .then(() => this.analytics.initialiseAnalytics())
-          .then(() => this.store$.dispatch(new StartSendingLogs()))
-          .then(() => this.navController.setRoot('JournalPage'))
-          .catch((error: AuthenticationError) => {
-            if (error === AuthenticationError.USER_CANCELLED) {
-              this.analytics.logException(error, true);
-            }
-            this.authenticationError = error;
-          }),
-      )
+    .then(() => {
+      this.authenticationProvider
+      .login()
+      .then(() => this.appConfigProvider.loadRemoteConfig())
+      .then(() => this.analytics.initialiseAnalytics())
+      .then(() => this.store$.dispatch(new StartSendingLogs()))
+      .then(() => {
+        const validDevice = this.deviceProvider.validDeviceType();
```

Comment:
This could be moved out to a function so that we keep this login function as lean as possible.

Reply:
Great suggestion. I had done this yesterday, but had to revert and reapply and missed doing this. Have changed code now.